hips/hip-0012.md

```diff
@@ -1,38 +1,138 @@
-Further criticism was that it was unclear who was "running the show" and how they were making decisions.
-Finally, we were criticized for not explaining the criteria we were using for determining when Helm 3 was complete.
-This document is intended to address these criticisms.
-It describes the process in detail, providing both a method for us to follow and a point of reference for community members to read.
+During the Helm 3 development process, Helm core maintainers received criticism that the process was not transparent enough.
+This proposal aims to outline a formal process for a successful Helm 4 release to address those criticisms.
+
+Further criticism included a lack of clarity regarding who was in charge and how decisions were being made, as well as uncertainty about the criteria used for determining when Helm 3 was complete.
+
+This document is intended to address these criticisms by presenting a clear plan for the timely delivery of a valuable Helm 4 release to the community.
+It describes the process in detail, providing both a roadmap for us to follow and a point of reference for community members to use when contributing changes.
+
+The process is based on the following ideas:
+- Requiring changes to be documented, either as “Helm 4 proposals” or standard HIPs, allows both Helm community members and maintainers to review them.
+This oversight ensures that changes are in line with this HIP, regarding their functional and architectural implementation.
+Additionally, documented changes enable contributors to independently implement agreed-upon proposals in coordination with the release engineer.
+- Requiring a timeboxed development effort intends to promote focus on the most valuable changes that the community and maintainers can deliver within a reasonable timeframe.
+It also aims to prevent an "open-ended" development cycle that might drag on or fizzle out.
+Ideally, the majority of changes will be proposed and agreed upon during the planning phase of the development cycle.
+However, amendments to the initial scope are permitted as long as they can fit within the agreed-upon timeframe.
+Explicitly allowing for scope amendments ensures that any changes are well thought out.
+For changes that don't make it into Helm 4, there will be a Helm 5 release of course!
 
 ## Specification
 
+### The process / timelines
+
+Helm 4 development will consist of three main phases:
+1. Planning
+1. Feature development
+1. Release preparation
+
+The timeline will be approximately:
+
+1. Marking this HIP as "accepted"
+1. Naming of a Release Engineer: Nov. 2024
+1. Kick-off meeting: KubeCon/CloudNativeCon North America 2024 (Salt Lake City)
+1. Earliest that engineering can begin: Nov. 2024
+1. Expected release of Helm v4.0.0 will be planned for the week before KubeCon/CloudNativeCon North America 2025 (expected November 2025)
```

Comment:
I think it is okay to tie to the date. Since the first Item is to "mark this HIP as accepted". Which should only be done if the dates at that point are in good order.

Agree, we can use "by" to indicate an earlier release can happen. Will adjust